### DISCUSSION TITLE
Remotecontrol_Power_Control not working with the newest firmware 1.33/1.31

### DISCUSSION THREAD

**Discussion by Jan1Scholz** [wills106/homeassistant-solax-modbus#488] [General]
Hallo . I have been using Remotecontrol_Power_Control emulating  Feedin Priority in the morning (till 12 am) and it worked for 4 months without any problem with firmware 1.30/1.28. Yesterday Solex automatically upgraded firmware to 1.33/1.31 a during Feedin Priority emulation the PV production is strange (constant, bellow possible production), see 
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/131488177/dad200c1-b324-40b8-94f7-672f30ac7642)
Do you have the same problem ? I have already downgraded firmware to 1.30/1.28.
Thank you,
Jan

**Comment by infradom** [comment]
Seems like you export control user limit was low (different scaling) with
the new firmware. I will check my firmware version tomorrow.

Op do 10 aug. 2023 18:45 schreef Jan1Scholz ***@***.***>:

**Comment by infradom** [comment]
Maybe the hidden control config_export_control_limit_readscale can be used to correct this.
We introduced this entitiy because some software versions had a bad scaling of the export_control_user_limit by a factor 10 
At what time did you revert to the old software version ? Was that around 12 ? 
I am a bit confused about the period in which the new firmware was active.

**Reply by Jan1Scholz** [reply]
I reverted to the old version in the evening (around 19:00) - the whole graph is with the new software. Constant production was when the feed-in priority was emulated by remotecontrol.

**Reply by Jan1Scholz** [reply]
The latest firmware for USB update is here: https://clubhouse.iotcz.cz/FVE/Firmware-SolaX/X3-Hybrid-G4_1.33-1.31/?fbclid=IwAR3UwuVhTZlJIzyf8OujnLLdLYJyGxx-8k_qkOQ1kSEfbAvhe171KqO7gwA All inertors will be automatically updated by Solax due to critical bug in firmware in the older versions, which can destroy the invertor.

**Reply by infradom** [reply]
Thanks, that clarifies things better; the bad news is that I have no explanation anymore. I do have seen that sometimes the system enters into a somewhat strange mode for a couple of hours (battery rebalancing ?) , so observing over a longer period might help.
Could you also explain what the graph shows: is it the PV power or the feedin power or something else?

For your information: emulated feedin priority computes the active power as:
```
        if pv > houseload_nett:  ap_target = 0 - pv + (houseload_brut - houseload_nett)*1.20  # 0 - pv + (houseload_brut - houseload_nett)
        else:                    ap_target = 0 - houseload_nett
```
where:
```
    meas           = datadict.get('measured_power', 0)
    pv             = datadict.get('pv_power_total', 0)
    houseload_nett = datadict.get('inverter_load', 0) - meas
    houseload_brut = pv - datadict.get('battery_power_charge', 0) - meas
```

Regarding software versions: then numbers you mention do not ring a bell. I only can see what my inverter says:
On the display:
DSP: 2.07
ARM: 2.03
And in HA:

<img width="313" alt="image" src="https://github.com/wills106/homeassistant-solax-modbus/assets/11804014/468a3f9e-9d5c-4f3a-9209-bec5910b14cb">

**Reply by Jan1Scholz** [reply]
The graph is PV power.  Your firmware version is 1.23/1.24, you can see it also on the inverter´s display (about invertor code)
![IMG_20230811_082721](https://github.com/wills106/homeassistant-solax-modbus/assets/131488177/b380d1ba-b7df-4597-97eb-b290c41525cb) my current version is 1.30/1.28.

**Reply by infradom** [reply]
OK, I will have to schedule a software update to try to reproduce this. My inverter is normally not connected to the cloud, so I need to open the firewall and ask for an update. May take some time my installer finds some time to do this..

**Reply by Jan1Scholz** [reply]
you can upgrade your invertor via USB flash disk: https://www.manualslib.com/manual/1319376/Solax-X3-Hybrid.html?page=23. The latest firmware for USB update is here: https://clubhouse.iotcz.cz/FVE/Firmware-SolaX/X3-Hybrid-G4_1.33-1.31/?fbclid=IwAR3UwuVhTZlJIzyf8OujnLLdLYJyGxx-8k_qkOQ1kSEfbAvhe171KqO7gwA

**Reply by infradom** [reply]
Thanks, I will try to do this during daytime tomorrow

**Reply by Jan1Scholz** [reply]
USB flash disk must be FAT32 formatted.  The file must be in the following directories: X:\update\ARM\binary_ARM_1.31.xyz.usb
X:\update\DSP\binary_DSP_1.33.xyz.usb
ARM is updated firstly, DSP secondly. 
I have updated firmware cca 7 times and it worked without any problem .... in 5 minutes is everything finished :-)

**Reply by infradom** [reply]
Sorry for the delay, I did not find time today, maybe tomorrow

**Reply by infradom** [reply]
Sorry again, I have caught a bad flue and have been in bed the past days

**Comment by autacek** [comment]
I have a similar problem since yesterday, I can't switch modes in HA, it always returns to the original settings within a few seconds, I can only switch manually from the solax application. What is wrong?
![Screenshot_20230811_090300_Home Assistant](https://github.com/wills106/homeassistant-solax-modbus/assets/115377236/91e04113-30de-4e3c-b9a0-7a50dae34796)

**Reply by autacek** [reply]
May be i found the reason.
![image](https://github.com/wills106/homeassistant-solax-modbus/assets/115377236/45abc2aa-6ccf-4f37-9dab-4e1dfb24398e)

**Reply by wills106** [reply]
> Inverter never connected to the Cloud / possibly after a firmware update: (SolaX only)
> 
> If you can read values, but unable to adjust select / number you need to press the "Unlock Inverter" button.
> Might need performing again following a full Power Cycle

Or do you have an automation triggering a change to the lock select?

**Comment by frank8the9tank** [comment]
Is my issue related to this? #492

**Reply by infradom** [reply]
As far as I can judge, this issue has the same cause as #492
